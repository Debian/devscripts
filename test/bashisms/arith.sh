#!/bin/sh
metric=0
echo $((metric=metric+1))

m=0
n=2
echo $((n-m++)) # BASHISM
echo $((++m))   # BASHISM
echo $(( m-- )) # BASHISM
echo $((--m))   # BASHISM

foo_bar=0
echo $((foo_bar++)) # BASHISM
echo $((foo_bar=foo_bar*2))
echo $((foo_bar*3/6))

echo $((2*n++)) # BASHISM

echo $(($n*n++)) # BASHISM

echo $((3**2)) # BASHISM
