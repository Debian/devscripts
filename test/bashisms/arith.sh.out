possible bashism in bashisms/arith.sh line 7 ('$((n++))' should be '$n; $((n=n+1))'):
echo $((n-m++)) # BASHISM
possible bashism in bashisms/arith.sh line 8 ('$((++n))' should be '$((n=n+1))'):
echo $((++m))   # BASHISM
possible bashism in bashisms/arith.sh line 9 ('$((n--))' should be '$n; $((n=n-1))'):
echo $(( m-- )) # BASHISM
possible bashism in bashisms/arith.sh line 10 ('$((--n))' should be '$((n=n-1))'):
echo $((--m))   # BASHISM
possible bashism in bashisms/arith.sh line 13 ('$((n++))' should be '$n; $((n=n+1))'):
echo $((foo_bar++)) # BASHISM
possible bashism in bashisms/arith.sh line 17 ('$((n++))' should be '$n; $((n=n+1))'):
echo $((2*n++)) # BASHISM
possible bashism in bashisms/arith.sh line 19 ('$((n++))' should be '$n; $((n=n+1))'):
echo $(($n*n++)) # BASHISM
possible bashism in bashisms/arith.sh line 21 (exponentiation is not POSIX):
echo $((3**2)) # BASHISM
